python/cudf/cudf/core/index.py

```diff
@@ -69,8 +66,41 @@
 T = TypeVar("T", bound="Frame")
 
 
+class IndexMeta(type):
+    """Custom metaclass for Index that overrides instance/subclass tests."""
+
+    def __call__(cls, data, *args, **kwargs):
+        if cls is Index:
+            return as_index(
+                arbitrary=data,
+                *args,
+                **kwargs,
+            )
+        return super().__call__(data, *args, **kwargs)
+
+    def __instancecheck__(self, instance):
+        if self in {
+            cudf.CategoricalIndex,
+            cudf.TimedeltaIndex,
+            cudf.DatetimeIndex,
+            cudf.IntervalIndex,
+        }:
+            return False
+        return isinstance(instance, BaseIndex)
```

Comment:
Would it be simpler to invert the condition? `if self is cudf.Index` rather than having to list all the different child classes?

Reply:
Done 👍